Fixed startup errors, removed/replaced deprecated packages, fixed test t...

jericsmith opened the pull request:
...o all run green, only cleaned up usings in files I touched

jericsmith committed:
Fixed startup errors, removed/replaced deprecated packages, fixed test to all run green, only cleaned up usings in files I touched

glennblock commented on src/WebApiBook.IssueTrackerApi/app_start/CollectionJsonConfig.cs:
```diff
@@ -0,0 +1,78 @@
+namespace WebApiBook.IssueTrackerApi.SampleApi
```
What is this file for? How does that relate to the PR?

glennblock commented on test/WebApiBook.IssueTrackerApi.AcceptanceTests/Features/CreatingIssues.cs:
```diff
@@ -13,7 +13,7 @@ namespace WebApiBook.IssueTrackerApp.AcceptanceTests.Features
     public class CreatingIssues : IssuesFeature
     {
-        private Uri _issues = new Uri("http://localhost/issue");
+        private Uri _issues = new Uri("http://localhost:8080/issue");
```
Port doesn't matter here as we're not sending anything across the network / everything is in memory. Why do you think the port needs to be there?

jericsmith replied:
If I remember correctly there was a test condition dependent upon it.   I'll take a look asap

--- Original Message ---

From: "Glenn Block" notifications@github.com
Sent: October 21, 2014 3:11 PM
To: "webapibook/issuetracker" issuetracker@noreply.github.com
Cc: "Eric Smith" j.eric.smith@live.com
Subject: Re: [issuetracker] Fixed startup errors, removed/replaced deprecated packages, fixed test t... (#47)

> @@ -13,7 +13,7 @@ namespace WebApiBook.IssueTrackerApp.AcceptanceTests.Features
>  {
>      public class CreatingIssues : IssuesFeature
>      {
> -        private Uri _issues = new Uri("http://localhost/issue");
> -        private Uri _issues = new Uri("http://localhost:8080/issue");

Port doesn't matter here as we're not sending anything across the network / everything is in memory. Why do you think the port needs to be there?

---

Reply to this email directly or view it on GitHub:
https://github.com/webapibook/issuetracker/pull/47/files#r19171675

jericsmith replied:
my bad

glennblock commented on test/WebApiBook.IssueTrackerApi.AcceptanceTests/app_start/CollectionJsonConfig.cs:
```diff
@@ -0,0 +1,78 @@
+namespace WebApiBook.IssueTrackerApp.AcceptanceTests.SampleApi
```
Why is this here?

jericsmith replied:
added with package updates

glennblock commented on src/WebApiBook.IssueTrackerApi/WebApiConfiguration.cs:
```diff
@@ -29,7 +29,7 @@ public static void Configure(HttpConfiguration config, IIssueStore issueStore =
         private static void ConfigureFormatters(HttpConfiguration config)
         {
-            config.Formatters.Add(new CollectionJsonFormatter());
+            config.Formatters.Add(new CollectionJsonFormatter.CollectionJsonMediaTypeFormatter());
```
This is really just a matter of preference. I prefer having the using rather than fully qualifying the references.

jericsmith replied:
with the CollectionJsonFormatter package update, CollectionJsonFormatter is a namespace. The test was failing.

glennblock commented:
Hi @jericsmith 

Thanks for the effort. I sent you a bunch of questions / comments.

jericsmith commented:
Ok.   I'll revisit the code asap to answer.  Thanks

--- Original Message ---

From: "Glenn Block" notifications@github.com
Sent: October 21, 2014 3:13 PM
To: "webapibook/issuetracker" issuetracker@noreply.github.com
Cc: "Eric Smith" j.eric.smith@live.com
Subject: Re: [issuetracker] Fixed startup errors, removed/replaced deprecated packages, fixed test t... (#47)

Hi @jericsmith

Thanks for the effort. I sent you a bunch of questions / comments.

---

Reply to this email directly or view it on GitHub:
https://github.com/webapibook/issuetracker/pull/47#issuecomment-59981755

jericsmith committed:
Advanced to Conflict Testing

glennblock commented:
This is sending too many changes in a single PR. Can you break out each significant change into a sep PR so that each PR only addresses one thing?

Thanks!
